### ISSUE TITLE
chat completion任务时输出大量<|EOT|> token

### ISSUE THREAD

**Issue by CarolXh** [deepseek-ai/DeepSeek-Coder#136]
使用官方的chat代码：

from transformers import AutoTokenizer, AutoModelForCausalLM
import torch
model_path = "/data/sonald/ai_models/model_weights/deepseek-coder-6.7b-instruct"
tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
model = AutoModelForCausalLM.from_pretrained(model_path, trust_remote_code=True, torch_dtype=torch.bfloat16).cuda()
messages=[
    { 'role': 'user', 'content': input[0]}
]
inputs = tokenizer.apply_chat_template(messages, add_generation_prompt=True, return_tensors="pt").to(model.device)
# tokenizer.eos_token_id is the id of <|EOT|> token
outputs = model.generate(inputs, max_new_tokens=3072, do_sample=True, top_k=50, top_p=0.95, num_return_sequences=1, eos_token_id=32014)
print(tokenizer.decode(outputs[0][len(inputs[0]):], skip_special_tokens=True))

然后发现当输出不满max_new_tokens数量时，会填充大量<|EOT|> token，似乎是skip_special_tokens这个参数没有生效？例如输出为：
        答案正确。
        <|EOT|>
        <|EOT|>
        <|EOT|>
        <|EOT|>
        <|EOT|>
        <|EOT|>
        <|EOT|>
        <|EOT|>
        <|EOT|>
        <|EOT|>
        <|EOT|>
         ......（一直到填满3072个）

我的输入input[0]是一段大约长度2k+的问题，其中包含了一些few-shot，格式类似：
###query：
###output：

会和这个有关系吗？此外运行时报出信息为：The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:32014 for open-end generation.

**Comment by guoday** [comment]
<|EOT|>是终止符号，eos_token_id为32021，不是32014

**Comment by guoday** [comment]
文档已经改成:
outputs = model.generate(inputs, max_new_tokens=512, do_sample=False, top_k=50, top_p=0.95, num_return_sequences=1, eos_token_id=tokenizer.eos_token_id)
